src/commands/command_parser.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cctype>
+#include <functional>
+#include <iterator>
+
+#include "parse_util.h"
+#include "status.h"
+#include "util.h"
+
+template <typename Iter>
+struct MoveIterator : Iter {
+  explicit MoveIterator(Iter iter) : Iter(iter){};
+
+  using Iter::value_type;
+
+  typename Iter::value_type&& operator*() const { return std::move(this->Iter::operator*()); }
+};
+
+template <typename Iter>
+struct CommandParser {
+ public:
+  using value_type = typename Iter::value_type;
+
+  CommandParser(Iter begin, Iter end) : begin(begin), end(end) {}
+
+  template <typename Container>
+  explicit CommandParser(const Container& con, size_t skip_num = 0)
+      : begin(std::begin(con) + skip_num), end(std::end(con)) {}
+
+  template <typename Container>
+  explicit CommandParser(Container&& con, size_t skip_num = 0)
+      : begin(MoveIterator(std::begin(con) + skip_num)), end(MoveIterator(std::end(con))) {}
+
+  decltype(auto) RawPeek() const { return *begin; }
+
+  decltype(auto) RawTake() { return *begin++; }
+
+  decltype(auto) RawNext() { ++begin; }
+
+  bool Good() const { return begin != end; }
+
+  template <typename Pred>
+  bool EatPred(Pred&& pred) {
+    if (Good() && std::forward<Pred>(pred)(RawPeek())) {
+      RawNext();
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  bool EatEqICase(std::string_view str) {
+    return EatPred([str](const auto& v) { return Util::EqualICase(str, v); });
```

Comment:
In some places, you have `&` near the type but in the `kvrocks` it's usually used near the argument's name. It's just about the style consistency across the project.

Reply:
done

Reply:
Oh, I changed it, but clang-format changed it back.